keentools_facebuilder/fbloader.py

```diff
@@ -456,42 +456,34 @@ def _unfix_not_in_groups(fb, head):
                     or not cam.auto_focal_estimation)
 
         def _auto_focal_estimation_mode_and_fixes():
-            mode = 'FB_ESTIMATE_VARYING_FOCAL_LENGTH'
             if head.smart_mode():
                 if camera.auto_focal_estimation:
                     if camera.is_in_group():
-                        for cam in head.cameras:
-                            fb.set_focal_length_fixed_at(
-                                cam.get_keyframe(),
-                                cam.image_group != camera.image_group
-                                or not cam.auto_focal_estimation)
-                        mode = 'FB_ESTIMATE_STATIC_FOCAL_LENGTH'
+                        fb.set_static_focal_length_estimation(settings.get_projection_matrix()[0][0])
                     else:  # image_group in (-1, 0)
+                        fb.set_varying_focal_length_estimation()
                         for cam in head.cameras:
                             fb.set_focal_length_fixed_at(
                                 cam.get_keyframe(),
                                 cam.image_group > 0
                                 or not cam.auto_focal_estimation)
-                        mode = 'FB_ESTIMATE_VARYING_FOCAL_LENGTH'
                 else:
+                    fb.set_varying_focal_length_estimation()
                     _unfix_not_in_groups(fb, head)
-                    mode = 'FB_ESTIMATE_VARYING_FOCAL_LENGTH'
             else:  # Override all
                 if head.manual_estimation_mode == 'all_different':
+                    fb.set_varying_focal_length_estimation()
                     _unfix_all(fb, head)
-                    mode = 'FB_ESTIMATE_VARYING_FOCAL_LENGTH'
                 elif head.manual_estimation_mode == 'current_estimation':
+                    fb.set_varying_focal_length_estimation()
                     _fix_all_except_this(fb, head, kid)
-                    mode = 'FB_ESTIMATE_VARYING_FOCAL_LENGTH'
                 elif head.manual_estimation_mode == 'same_focus':
-                    _unfix_all(fb, head)
-                    mode = 'FB_ESTIMATE_STATIC_FOCAL_LENGTH'
+                    fb.set_static_focal_length_estimation(settings.get_projection_matrix()[0][0])
```

Comment:
settings.get_projection_matrix()[0][0]?

Reply:
Not sure. Have a look at coords.projection_matrix(). There is a special scale factor dependent on camera orientation (and blender view fixes):

def projection_matrix(w, h, fl, sw, near, far, scale=1.0):
    z_diff = near - far
    fl_to_sw = fl / sw
    return np.array(
        [[scale * w * fl_to_sw, 0, 0, 0],
         [0, scale * w * fl_to_sw, 0, 0],
         [-w / 2, -h / 2, (near + far) / z_diff, -1],
         [0, 0, 2 * near * far / z_diff, 0]]
    ).transpose()

Reply:
coords.focal_by_projection_matrix(projection_matrix, sensor_width) looks like solving the problem